### DISCUSSION TITLE
Enhancement: use the app as a simplifed and cheap "proof-of-health"

### DISCUSSION THREAD

**Discussion by pe-pan** [covid19cz/erouska-android#402] [Ideas]
Let's say there is enough users of the app in population. Let's say that certain parties (an airport, planned surgery, dancing class, night bar, foreign contry, whatever) will require a proof-of-health (a negative test 2 days old max) from the user. If the app was collecting data (with the user agreement) for how long it was running (so the user is not powering it off or disabling bluetooth on purpose), the app could show a certificate that the user is using the app and not meeting infected people recently. This could serve to many parties as a cheap proof-of-health.

**Comment by pe-pan** [comment]
BTW, this could be also a motivational factor to install the app.

**Comment by davidvavra** [comment]
That would be nice :) Some people will be faking it (with a screenshot), but it's still worth it.

**Comment by pe-pan** [comment]
Give it an animation then. Perhaps current time/date + picture from the camera perhaps? Or, generate a 2D bar code that would be unique and valid for couple of seconds/minutes only; the 3rd party would have to read it to learn if it's valid or not. I'm sure there will be a way...

**Comment by vandac** [comment]
what about a case when the app shows I have not met anyone infected, yet the day after one of my recent contact gets tested positively? In such cases the app would work as a false safety proof.

**Comment by pe-pan** [comment]
The delay between meeting someone infected and learning about this fact is IMHO caused by the late entering of the code from the infected person. This is a crucial property of the app; not sure if code entering can be enforced or even automated (once the lab learns someone is positive, it will enter the code). I don't think so as the app does not collect any data identifying the person. On the other hand, late entering of the code is not breaking purpose of this feature only; it's actually breaking purpose of whole the app. So if we believe the application makes sense, then this feature must makes sense.

BTW, my role is not to analyze how to implement the feature, what are the risks or consequences, or which other requirements must be met first. I believe that more involved participants will analyze properly :).

**Comment by pe-pan** [comment]
BTW, I'm not an epidemiologist but what I remember... getting infected and going for a test at the very same moment does not mean the test will result to be positive. It will take some time before the infection gets evolved and during all this time the tests will give you false negative results. There is no perfect tool.
